Barber Update

Fox-McCloud opened the pull request:
So, it always irked me that Barber was never finished...likewise, it's essentially just a karma job for being able to play Civilian....so...this changes all of that. This is a major addition to the barber job

- Adds in scissors.
 - This is mostly Urist's scissors code. If you're on help intent and click someone, you can change their facial hair and regular hair (can only change facial hair if they're male OR Vulpkanin, of course). Only the barber will have scissors. I added cutting sounds to it for more fun and such.
 - There's a traitor job-specific version of these scissors that will cut someone's throat instead of cutting their hair. The traitor version does 18 damage (instead of 5). When you start cutting someone's hair with these it'll instead cut their throat and apply 18 damage while causing 30 oxygen damage over the next 20 seconds...it also causes...well...lots of blood. They'll set you back 5 TC.
 - Thanks to PseudoUnicorn for the name suggestion on the traitor scissors =p

- Adds in the Dye Generator and Dye Bottle
 - The dye generator is a simple machine that, when interacted with, will allow you to select just about any color you can possibly imagine. If you use the dye bottle on the dye generator then the dye bottle will have the same dye color as the generator.
 - The Dye bottle can be used on human mobs; when that's done, it'll give you the option of dying their hair, facial hair, or body color--please note, you can only change someone's facial hair color if they're male or Vulpkanin and you can only change body color if that's *normally* a thing for them (ie: Unathi, Vulps, Tajars, etc). 
 - As with the scissors, ONLY the barber will have this special refillable bottle.

- The barber will now start off with a barber kit
 - The kit contains a pair of barber scissors, the special hair dye bottle, a 50 unit bottle of hairgrownium, a 50 unit bottle of quantum hair dye, a spare empty bottle, a dropper, and a fake moustache.

- The barber will now start off with a lipstick kit
 - The kit contains 7 different colors of lipstick

- Gives the Barber alt titles
 - Hair Stylist and Beautician

Re-designs the Barber's area:

![barber](https://i.gyazo.com/92c974045f8af51c89e859518494c1e6.png)
Big thanks to @FullOfSkittles for the Hair Dye Generator and Hair Dye Bottle sprite. Thank you, Skittles!

Basically adds a potted plant, newscaster, and some "For looks" entertainment wall units, all in addition to a dresser....and the hair dye generator, of course.

I hope this improves the barber to a degree people will pick it because they want to at least somewhat do the job as opposed to just using it as civilian 2.0---not that there's anything wrong with the latter; it's still the perfect job for slacking off without hurting anyone.

Fox-McCloud commented:
started some basic work on the hair dye generator.

DavidRossii commented:
BARBER BECOMES.... USEFUL?! THIS IS AMAZING!
It'd be even more funky if the functionality of mirrors were removed (magic mirror changing capabilities) and I assume the mirror functionality applies to the scissors, same menu and such.

The day barber comes useful is a good day. :+1:

Fox-McCloud commented:
Nah, the scissors behave differently than the mirror; they use a list as opposed to buttons.

The issue I have with removing that ability from mirrors is that it makes that random who like to shave people's heads (or dump hairgrownium/super hairgronwium on people for the lulz) unable to correct those issue unless there's a barber on station. Still may be worth considering? I Dunno.

DavidRossii commented:
> The issue I have with removing that ability from mirrors is that it makes that random who like to shave people's heads (or dump hairgrownium/super hairgronwium on people for the lulz) unable to correct those issue unless there's a barber on station. Still may be worth considering? I Dunno. I kinda don't like that idea.

Oh gosh yeah that's true. But yeah, loving the added bonus to Barber, a job which has been seen as utterly useless, overall ood stuff.

Dumbdumn5 commented:
PRAISE BE UNTO BARBER! MOST IMPORTANT JOB! :+1:

Aurorablade commented:
PRAISE UNTO THE BARBER!

Fox-McCloud commented:
Ok, I did it.

Added in the dye generator along with its special hair dye bottle. See original description for details.

Spacemanspark commented:
This will be fun.
:+1:

Fox-McCloud commented:
Alright, to upate, basically I added in a dye machine that, in conjuction with a special item, allows you to dye someone's hair, facial hair, or body a different color---again, see OP for more details.

NTSAM commented:
yeeeeees :+1:

Wait, it lets you dye body color?
Think of all the fun madness. :smiling_imp:

Fox-McCloud commented:
>Wait, it lets you dye body color?

Yes, but only if you could already pick body color on your character--this doesn't allow for, say, humans that are pink, or Vox that are bright orange.

Robertha89 commented:
Bright Orange vox plz, also make the barber change humans skintones, cause that will probably not end well in any way ever.

But yes, Orange Vox Plz, would go to barber every round to be bright orange burd.

Fox-McCloud commented:
@veganzombeh 

The reason the barber can't do that is because it's already *not* possible with species you currently can't do it on. It would require an overhaul of the species appearance code for Vox (or others) in addition to a greyscaling of their sprites for that to be even possible.

Skin_tone mechanics operates on a different scale that's not inherently tied to hex color in any way, so I'd have to make a new item for that---don't really want to do that, in this particular PR, to be honest---maybe later.

Robertha89 commented:
I did consider greyscaling the vox sprites, buuut the original vox color would essentially be lost in the magic of color. Vox Hair styles can be fixed that way, cause that would mean you could have black quills for those fancy situations.

veganzombeh commented:
I think 10 TC may be a bit high. The scissors are very noticeable and don't incapacitate the victim. No reason to cost more than an esword or parapen IMO.

Fox-McCloud commented:
It's more the fact that they incur losebreath and are high-damage,but super easy to conceal when you're done with them, not to mention stun+scissors throat cutting is prettttty crazy because at 50 oxyloss you black out.

Fox-McCloud commented:
Updated the barber dye generator sprite, big thanks to @FullOfSkittles!

veganzombeh commented:
True, but in that time you could do more damage with two hits of an energy sword.

If they're intended to be a stealth weapon, I would suggest actually disguising them as standard scissors, and giving them some way to be used as standard scissors. Perhaps only have it cut throats on harm intent?

Fox-McCloud commented:
Added in the hair dye bottle sprite that @FullOfSkittles made---has overlays that will change based on the color of the dye in the bottle.

Also changed the price of the scissors from 10 to 5.

Fox-McCloud commented:
Ok, last thing:

- Made it so that the Dye Generator emits light, with the light color changing depending on the color of the hair dye selected.

creodas commented:
Old Henry will be a happy man

KasparoVy commented:
@Robertha89 If Vox hair/facial hair could be greyscaled and still look good, I'd be forever grateful.

Fox-McCloud commented:
Final touch: alt titles for the Barber: Hair Stylist and Beautician.

This is now ready to merge.

TheDZD commented on code/game/objects/items/weapons/scissors.dm:
```diff
@@ -0,0 +1,109 @@
+		//this is largely copypasted from there.
+		//handle facial hair (if necessary)
+		var/list/species_facial_hair = list()
+		if(H.gender == MALE || H.get_species() == "Vulpkanin")
```
In the future, we should probably implement some generic system for species to be able to have facial hair in both genders.

Fox-McCloud replied:
Definitely, because checking for male OR special species that has facial hair is annoying.

tigercat2000 commented:
Coooonflicts.

Fox-McCloud commented:
Fixed.

orelbon commented on code/game/jobs/job/support.dm:
```diff
@@ -362,6 +362,7 @@
 	spawn_positions = 1
 	supervisors = "the head of personnel"
 	selection_color = "#dddddd"
+	alt_titles = list("Hair Stylist","Beautician")
```
lol beautician, thanks for the laugh. i can't believe that's a real word.

TheDZD commented:
Cooooonfliiiiicts.
